web/src/lib/utils/focus-util.ts

```diff
@@ -1,4 +1,60 @@
-const selectors =
-  'button:not([disabled], .hidden), [href]:not(.hidden), input:not([disabled], .hidden), select:not([disabled], .hidden), textarea:not([disabled], .hidden), [tabindex]:not([tabindex="-1"], .hidden)';
+import { focusable, isTabbable, tabbable, type CheckOptions, type TabbableOptions } from 'tabbable';
 
-export const getFocusable = (container: ParentNode) => [...container.querySelectorAll<HTMLElement>(selectors)];
+type TabbableOpts = TabbableOptions & CheckOptions;
+let defaultOpts: TabbableOpts = {
+  includeContainer: false,
+};
+
+export const setDefaultTabbleOptions = (options: TabbableOpts) => {
+  defaultOpts = options;
+};
+
+export const getTabbable = (container: Element, includeContainer: boolean = false) =>
+  tabbable(container, { ...defaultOpts, includeContainer });
+
+export const focusNext = (selector: (element: HTMLElement | SVGElement) => boolean, forwardDirection: boolean) => {
+  const focusElements = focusable(document.body, { includeContainer: true });
+  const current = document.activeElement as HTMLElement;
+  const index = focusElements.indexOf(current);
+  if (index === -1) {
+    for (const element of focusElements) {
+      if (selector(element)) {
+        element.focus();
+        return;
+      }
+    }
+    focusElements[0].focus();
+    return;
+  }
+  if (forwardDirection) {
+    let i = index + 1;
+    while (i !== index) {
+      const next = focusElements[i];
+      if (!isTabbable(next) || !selector(next)) {
+        if (i === focusElements.length - 1) {
+          i = 0;
+        } else {
+          i++;
+        }
+        continue;
+      }
+      next.focus();
+      break;
+    }
+  } else {
+    let i = index - 1;
+    while (i !== index && i >= 0) {
+      const next = focusElements[i];
+      if (!isTabbable(next) || !selector(next)) {
+        if (i === 0) {
+          i = focusElements.length - 1;
+        } else {
+          i--;
```

Comment:
Probably just use a for loop from 0 to < focus element length. But set the next to i % length